File: NRZ.Data/Person.cs
using System;
using System.Collections.Generic;

namespace NRZ.Data
{
    public partial class Person
    {
        public Person()
        {
            AgriculturalMachinery = new HashSet<AgriculturalMachinery>();
            AuctionRegistration = new HashSet<AuctionRegistration>();
            SeizedPropertyAvailabilityRequestCheckedPerson = new HashSet<SeizedPropertyAvailabilityRequest>();
            SeizedPropertyAvailabilityRequestRequestorPerson = new HashSet<SeizedPropertyAvailabilityRequest>();
        }

        public int Id { get; set; }
        public string FirstName { get; set; }
        public string MiddleName { get; set; }
        public string LastName { get; set; }
        public string UserId { get; set; }
        public string CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; }
        public string UpdatedBy { get; set; }
        public DateTime? UpdatedOn { get; set; }
        public bool Deleted { get; set; }
        public string DeletedBy { get; set; }
        public DateTime? DeletedOn { get; set; }
        public string IdentificationNumber { get; set; }
        public string IdentificationNumberType { get; set; }
        public string Email { get; set; }
        public string Phone { get; set; }
        public int? AddressId { get; set; }

        public virtual Address Address { get; set; }
        public virtual AspNetUsers CreatedByNavigation { get; set; }
        public virtual AspNetUsers DeletedByNavigation { get; set; }
        public virtual IdentificationType IdentificationNumberTypeNavigation { get; set; }
        public virtual AspNetUsers UpdatedByNavigation { get; set; }
        public virtual AspNetUsers User { get; set; }
        public virtual ICollection<AgriculturalMachinery> AgriculturalMachinery { get; set; }
        public virtual ICollection<AuctionRegistration> AuctionRegistration { get; set; }
        public virtual ICollection<SeizedPropertyAvailabilityRequest> SeizedPropertyAvailabilityRequestCheckedPerson { get; set; }
        public virtual ICollection<SeizedPropertyAvailabilityRequest> SeizedPropertyAvailabilityRequestRequestorPerson { get; set; }
    }
}

File: NRZ.Web/client/src/models/person.ts
import Address from "@/models/address";

export default class Person {
    id: number | null;
    firstName: string;
    middleName: string;
    lastName: string;
    phone: string;
    email: string;
    identificationNumber: string;
    identificationType: string | null;
    userId: string | null;
    address?: Address | null;

    constructor(obj: any = {}) {
        this.id = obj.id || null;
        this.firstName = obj.firstName || '';
        this.middleName = obj.middleName || '';
        this.lastName = obj.lastName || '';
        this.phone = obj.phone || '';
        this.email = obj.email || '';
        this.identificationNumber = obj.identificationNumber || '';
        this.userId = obj.userId || null;
        this.identificationType = obj.identificationType || null;
        this.address = obj.address ? new Address(obj.address) : null;
    }
}

export class RegixPersonModel {
    constructor(obj: any = {}) {
        this.id = obj.id || 0;
        this.identifier = obj.identifier;
        this.firstName = obj.firstName;
        this.middleName = obj.middleName;
        this.lastName = obj.lastName;
        this.dateOfBirth = obj.dateOfBirth ? new Date(obj.dateOfBirth) : undefined;
        this.dateOfDeath = obj.dateOfDeath ? new Date(obj.dateOfDeath) : undefined;
        this.fullName = `${this.firstName ?? ''} ${this.middleName ?? ''} ${this.lastName ?? ''}`;
        this.requestId = obj.requestId || 0;
    }

    id: number;
    identifier: string;
    firstName: string;
    middleName: string;
    lastName: string;
    dateOfBirth?: Date;
    dateOfDeath?: Date;
    fullName: string;
    requestId?: number;
}
